src/controllers/krate/publish.rs

```diff
@@ -149,6 +153,11 @@ pub fn publish(req: &mut dyn Request) -> CargoResult<Response> {
         )?
         .save(&conn, &new_crate.authors, &verified_email_address)?;
 
+        if let AuthenticationSource::ApiToken(token_id) = req.authentication_source()? {
```

Comment:
What should I be doing in the else block? Getting the token in some other way? Returning an unauthorized error?